### PR TITLE
Add enableCrossZoneUpgrade and prioritizeUnhealthyInstances properties to RollingUpgradePolicy for VMSS

### PR DESCRIPTION
**Pull request by kimberlyyipc**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Note: These changes have been reviewed as **not breaking**

### CODE REVIEW

**Comment by kimberlyyipc** [comment]
The PR for these changes are still in review and I will provide update when our changes are merged to CRP

**Comment by kimberlyyipc** [comment]
I'm not sure why, but I cannot add tags to this PR. I would like to add the [DoNotMerge] label.

**Comment by kimberlyyipc** [comment]
@ajjha-az for review

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json [hunk 1]
```diff
@@ -9154,6 +9154,14 @@
...
           "type": "string",
           "description": "The wait time between completing the update for all virtual machines in one batch and starting the next batch. The time duration should be specified in ISO 8601 format. The default value is 0 seconds (PT0S)."
+        },
+        "enableCrossZoneUpgrade": {
+          "type": "boolean",
+          "description": "Enable rolling upgrade across multiple zones during UD walk within the limit of MaxBatchInstanceCount or BatchInstancePercent"
+        },
+        "prioritizeUnhealthyInstances": {
+          "type": "boolean",
+          "description": "Enable rolling upgrade to start with the batch which has most unhealthy VMs in percentage. If enableCrossZoneUpgrade is not enabled, the rolling upgrade will start from the most unhealthy availability zone,and within the zone, walk the UD that has the most unhealhty VMs first"
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json:9164]: 'unhealhty' should be 'unhealthy'.

**Review by Sandido — CHANGES_REQUESTED** [review]

**Review by pilor — APPROVED** [review]

**Comment by fengzhou-msft** [comment]
@kimberlyyipc please address the requested change.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json [hunk 2]
```diff
@@ -9154,6 +9154,16 @@
         "pauseTimeBetweenBatches": {
           "type": "string",
           "description": "The wait time between completing the update for all virtual machines in one batch and starting the next batch. The time duration should be specified in ISO 8601 format. The default value is 0 seconds (PT0S)."
+        },
+
+        "enableCrossZoneUpgrade": {
+          "type": "boolean",
+          "description": "Enable rolling upgrade across multiple zones during UD walk within the limit of MaxBatchInstanceCount or BatchInstancePercent"
```
Comment (ajjha-az) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json:9557]: "description": Allow VMSS to ignore AZ boundaries when constructing upgrade batches. Take into consideration Update Domain and maxBatchInstancePercent to determine the batch size.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json [hunk 3]
```diff
@@ -9154,6 +9154,16 @@
...
+        },
+
+        "prioritizeUnhealthyInstances": {
+          "type": "boolean",
+          "description": "Enable rolling upgrade to start with the batch which has most unhealthy VMs in percentage. If enableCrossZoneUpgrade is not enabled, the rolling upgrade will start from the most unhealthy availability zone,and within the zone, walk the UD that has the most unhealhty VMs first"
```
Comment (ajjha-az) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json:9561]: "description": Upgrade all unhealthy instances in a scale set before any healthy instances

Comment (kimberlyyipc) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json]: done

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json (hunk 2 above)
Comment (kimberlyyipc) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json]: done

**Comment by kimberlyyipc** [comment]
These changes have been reviewed as not breaking.

**Review by Sandido — APPROVED** [review]

**Comment by kimberlyyipc** [comment]
@fengzhou-msft Hi, can you please take a look at my changes again? Also, how can I go about resolving the SDK azure-sdk-for-go and SDK azure-sdk-for-net failures in the pipeline?

**Comment by PhoenixHe-msft** [comment]
The Go SDK error is a known issue, you can ignore it now. The .NET SDK error is test project build failure which is expected now. Safe to ignore.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**